### ISSUE TITLE
Have you figured out why tensorflow is slower than pytorch? It's so wired.

### ISSUE THREAD

**Issue by XinJiang1994** [PeterL1n/BackgroundMattingV2-TensorFlow#5]

**Comment by XinJiang1994** [comment]
I run the test too. I use a tesla v100. Tensorflow takes 1s on average for an image with tf.fuction and 0.2s without tf.function. But pytorch only need 0.02s. This is very puzzling.

**Comment by PeterL1n** [comment]
My experience in general is that TensorFlow in general just bad... The tensor is arranged in NHWC format where as most GPU are native for NCHW. Even though you can implement it in NCHW for convolution and batchnorm etc, but some operators such as resizing still only support NHWC, so there will be a lot of extra transposing. TensorFlow also doesn't seem to support tensor views like PyTorch, so every reshape is a copy. PyTorch is a lot more flexible.
